src/config.rs

```diff
@@ -163,6 +163,19 @@ pub trait Config: Default + PartialEq + DeserializeOwned + Serialize {
     ///
     /// Return false if the configuration does not support storing the migration version
     fn set_migration_version(&mut self, _version: u32) -> bool;
+
+    fn list_available_config(&self) -> &'static [ConfigField];
+}
+
+#[derive(Serialize)]
+pub struct ConfigField {
+    pub name: &'static str,
```

Comment:
rename to one letter names?  that would allow us to support more fields before reaching the message size limit.